sdk/ai/ai-inference-rest/src/tracingPolicy.ts

```diff
@@ -0,0 +1,252 @@
+        throw err;
+      }
+    },
+  };
+}
+
+function tryCreateTracingClient(): TracingClient | undefined {
+  try {
+    return createTracingClient({
+      namespace: "",
+      packageName: "@azure-rest/ai-inference",
+      packageVersion: SDK_VERSION,
+    });
+  } catch (e: unknown) {
+    logger.warning(`Error when creating the TracingClient: ${getErrorMessage(e)}`);
+    return undefined;
+  }
+}
+
+function tryCreateSpan(
+  tracingClient: TracingClient,
+  request: PipelineRequest,
+  spanAttributes: Record<string, unknown>,
+): { span: TracingSpan; tracingContext: TracingContext } | undefined {
+  try {
+    // As per spec, we do not need to differentiate between HTTP and HTTPS in span name.
+    const { span, updatedOptions } = tracingClient.startSpan(
+      "{gen_ai.operation.name} {gen_ai.request.model}", ///TODO: not sure how to retrieve the name yet
+      { tracingOptions: {} },  ///TODO: should I pass tracing options?
+      {
+        spanAttributes,
+      },
+    );
+
+    // If the span is not recording, don't do any more work.
+    if (!span.isRecording()) {
+      span.end();
+      return undefined;
+    }
+
+    // set headers
+    const headers = tracingClient.createRequestHeaders(
+      updatedOptions.tracingOptions.tracingContext,
+    );
+    for (const [key, value] of Object.entries(headers)) {
+      request.headers.set(key, value);
+    }
+    return { span, tracingContext: updatedOptions.tracingOptions.tracingContext };
+  } catch (e: any) {
+    logger.warning(`Skipping creating a tracing span due to an error: ${getErrorMessage(e)}`);
+    return undefined;
+  }
+}
+
+function tryProcessError(span: TracingSpan, error: unknown): void {
+  try {
+    span.setStatus({
+      status: "error",
+      error: isError(error) ? error : undefined,
+    });
+    span.end();
+  } catch (e: any) {
+    logger.warning(`Skipping tracing span processing due to an error: ${getErrorMessage(e)}`);
+  }
+}
+
+function tryProcessResponse(span: TracingSpan, response: PipelineResponse): void {
+  try {
+    const body = JSON.parse(response?.bodyAsText || "{}");
+    /* sample of body for success
+    const body = {
+      choices: [
+        {
+          finish_reason: "stop",
+          index: 0,
+          message: {
+            content: " The capital of France is Paris.",
+            role: "assistant",
+            tool_calls: [
+            ],
+          },
+        },
+      ],
+      created: 1723767112,
+      id: "cmpl-53909b0bebf94dab83215ffd99acfd95",
+      model: "phi3-medium-128k",
+      object: "chat.completion",
+      usage: {
+        completion_tokens: 8,
+        prompt_tokens: 10,
+        total_tokens: 18,
+      },
+    };
+     */
+    /* sample of body for bad model name
+    const body = {
+      error: {
+        code: "unknown_model",
+        message: "Unknown model: chat-gpt-large",
+        details: null,
+      },
+    };
+     */
+    span.setStatus({
+      status: "success",
```

Comment:
we should not set status to success, let's leave it unset

Reply:
Unset now